simplicity-sys/src/c_jets/c_frame.rs

```diff
@@ -0,0 +1,54 @@
+//! High level APIs to creating read/write CFrames
+//!
+
+use std::os::raw::c_uchar;
+
+use super::frame_ffi;
+use super::frame_ffi::CFrameItem;
+
+impl CFrameItem {
+    /// Allocate a new frame item with dummy values
+    unsafe fn new_unchecked() -> Self {
+        Self {
+            edge: std::ptr::null(),
+            len: 0,
+        }
+    }
+
+    /// Initialize a new read frame.
+    /// 'n' is the number of cells for the read frame.
+    /// 'from' is a pointer to the beginning of the new slice for the array of u8 to hold the frame's cells.
+    ///
+    /// Note: The C implementation uses array of UWORD for `from`. We use a slice of u8 instead.
+    pub fn new_read(n: usize, from: *const c_uchar) -> Self {
```

Comment:
In 59c2942084530ec7c2d06b9147dccd3156aa51c3:

This is definitley not a safe function if we take a `*const c_uchar` because it is possible to make such a pointer which is unaligned when reinterpreted as a `*UWORD`.

It is also definitely not safe because it calls into `initReadFrame` which has as a precondition that `from` points to an array of `n` `UWORD`s, while this API allows it to be null or dangling or to point to uninitialized memory.